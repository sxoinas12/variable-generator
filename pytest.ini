[pytest]
testpaths=
    word_processor/tests
filterwarnings =
    ignore::RuntimeWarning
    ignore::DeprecationWarning
