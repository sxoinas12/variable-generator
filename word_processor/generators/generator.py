import spacy
from py.error import Error

nlp = spacy.load("en_core_web_trf")


class Generator():
    """
        Generator class that will use a strategy resolver to generate suggestions for a text
        :return:

    """

    # passed text
    text = None
    # parsed text from spacy lib
    doc = None
    # the strategy resolver to use
    strategy_resolver = None
    # the parser for the presentation
    parser = None

    # suggestions calculated
    suggestions = None

    def __init__(self, text=None, strategy=None, parser=None):
        self.strategy_resolver = strategy
        self.parser = parser
        if text is not None:
            self.set_text(text)

    def suggest(self):
        if self.doc is None or self.text is None:
            raise Error('No document')

        self.suggestions = self.strategy_resolver(self.doc)

        return self.suggestions

    def set_text(self, text):
        self.text = text
        self.doc = nlp(text)

    def set_strategy(self, strategy_resolver):
        self.strategy_resolver = strategy_resolver

    def present(self):
        self.parser.suggestions = self.suggestions
        return self.parser.present()