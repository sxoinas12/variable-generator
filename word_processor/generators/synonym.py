import nltk
from nltk.corpus import wordnet
import os

from nltk.corpus.reader import WordNetError

print("Loading nltk", os.path.dirname(__file__) + '/../../nltk_data/')
nltk.data.path.append(os.path.dirname(__file__) + '/../../nltk_data/')


def get_synonyms(word):
    synonyms = []

    try:
        synsets_words = wordnet.synsets(word)
    except WordNetError:
        return []

    for syn in synsets_words:
        for lemma in syn.lemmas():
            synonyms.append(lemma.name())

    return synonyms


def get_antonyms(word):
    antonyms = []

    try:
        synsets_words = wordnet.synsets(word)
    except WordNetError:
        return []

    for syn in synsets_words:
        for lemma in syn.lemmas():
            if lemma.antonyms():
                for antonym in lemma.antonyms():
                    antonyms.append(antonym.name())

    return antonyms