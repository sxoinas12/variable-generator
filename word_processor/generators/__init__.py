from .generator import Generator
from .synonym import get_synonyms, get_antonyms

__all__ = [
    'Generator',
    'get_antonyms',
    'get_synonyms',
]
