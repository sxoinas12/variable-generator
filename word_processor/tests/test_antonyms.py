import json

import pytest

import word_processor.generators.synonym as synonym


@pytest.mark.parametrize(
    'word',
    [
        'dog',
        'cat',
        'fetcher',
        'serializer',
        'factory',
    ]
)
def test_antonyms(snapshot, word):
    result = synonym.get_antonyms(word)

    assert result == snapshot