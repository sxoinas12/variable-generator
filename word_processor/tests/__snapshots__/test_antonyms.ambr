# name: test_antonyms[cat]
  <class 'list'> [
  ]
---
# name: test_antonyms[dog]
  <class 'list'> [
  ]
---
# name: test_antonyms[factory]
  <class 'list'> [
  ]
---
# name: test_antonyms[fetcher]
  <class 'list'> [
  ]
---
# name: test_antonyms[serializer]
  <class 'list'> [
  ]
---
